dictdatabase/io_unsafe.py

```diff
@@ -79,21 +88,12 @@ def partial_read_only(db_name: str, key: str) -> dict | None:
 
 	# Not found in index file, search for key in the entire file
 	all_file_bytes = io_bytes.read(db_name)
-	key_start, key_end = utils.find_outermost_key_in_json_bytes(all_file_bytes, key)
-
-	if key_end == -1:
+	start, end, found = Searcher().search(all_file_bytes, key)
```

Comment:
Also, no class required since there is no internal state. 
By the way, The overhead is about 60ns, which is double of a function call

Reply:
Ok, but many parts are difficult to understand due to the fact that they are divided into many calls. How about replacing the Search classes and Index Manager with module files that will have all the logic with searching/building indexes?

for example:
```
key_start, key_end = utils.find_outermost_key_in_json_bytes(file, key)
if key_end == -1:
    return -1, -1, False
start = key_end + (1 if file[key_end] == byte_codes.SPACE else 0)
end = utils.seek_index_through_value_bytes(file, start)
```
        
it is difficult to understand that the beginning and end of the value of the key itself will be obtained here

`from searcher import get_value_by_key` 
in which this logic  - will be easier for understanding

Reply:
Okay, that sounds good!